fix, on ppc64le and RHEL / CentOS 8, for: identifier "__ieee128" is undefined
See https://github.com/easybuilders/easybuild-easyconfigs/issues/11913
Patch by Simon Branford (University of Birmingham)
--- cmake/Dependencies.cmake.orig
+++ cmake/Dependencies.cmake
@@ -1537,6 +1537,9 @@
   if(CMAKE_POSITION_INDEPENDENT_CODE AND NOT MSVC)
     list(APPEND CUDA_NVCC_FLAGS "-Xcompiler" "-fPIC")
   endif()
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "(powerpc|ppc)64le")
+    list(APPEND CUDA_NVCC_FLAGS "-Xcompiler" "-mno-float128")
+  endif()
 
   if(CUDA_HAS_FP16 OR NOT ${CUDA_VERSION} LESS 7.5)
     message(STATUS "Found CUDA with FP16 support, compiling with torch.cuda.HalfTensor")
--- third_party/gloo/cmake/Cuda.cmake.orig
+++ third_party/gloo/cmake/Cuda.cmake
@@ -169,6 +169,9 @@
   gloo_list_append_if_unique(CUDA_NVCC_FLAGS "-std=c++11")
 endif()
 gloo_list_append_if_unique(CUDA_NVCC_FLAGS "-Xcompiler" "-fPIC")
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "(powerpc|ppc)64le")
+  gloo_list_append_if_unique(CUDA_NVCC_FLAGS "-Xcompiler" "-mno-float128")
+endif()
 
 mark_as_advanced(CUDA_BUILD_CUBIN CUDA_BUILD_EMULATION CUDA_VERBOSE_BUILD)
 mark_as_advanced(CUDA_SDK_ROOT_DIR CUDA_SEPARABLE_COMPILATION)
